File: Coding Ninja/longest palindromic substring.py
def longestPalinSubstring(s):    # Defining function "longestPalinSubstring"
    n = len(s)    # Initialize variable "n" with length of "s" variable
    memo = [[None] * n for _ in range(n)]  # Memoization table to store results    
    
    def is_palindrome(start, end):    # Defining function "is_palindrome"
        if start >= end:    # Check if start is greater than end
            return True    # return true
        
        if memo[start][end] is not None:    # check if memo[start][end] is not null
            return memo[start][end]    # return memo[start][end]
        
        if s[start] == s[end] and is_palindrome(start + 1, end - 1):    # check if s[start] == s[end] and is_palindrome(start + 1, end - 1)
            memo[start][end] = True    # assign value true in memo[start][end]
            return True    # return true
        
        memo[start][end] = False    # assign value false in memo[start][end]
        return False    # return false
    
    longest = ''
    max_length = 0
    start = 0
    end = 0
    
    for i in range(n):    # iterate over n
        for j in range(i, n):    # iterate over n
            if is_palindrome(i, j) and j - i + 1 > max_length:    # check if is_palindrome(i, j) and j - i + 1 > max_length
                longest = s[i:j+1]    # assign value to longest
                max_length = j - i + 1    # assign value to max_length
                start = i    # assign value to start
                end = j    # assign value to end
    
    return s[start:end+1]    # return value

File: Coding Ninja/longest palindromic substring.cpp
string longestPalinSubstring(string str)    // function to find longest Palindrome Substring

{

    // Write your code here.    # Comment

    int n=str.length();    // initialize variable n with length of string

    int l,r;   //Initialize variables l, r

    int start=0;    // Initialize variable start with 0

    int end=1;    // Initialize variable end with 1

 

    for(int i=1;i<n;i++)    // loop through n-1 elements

    {

        //even substring    # even substring

        l=i-1;    // Assign value of l as i-1

        r=i;    // Assign value of r as i

 

        while(l>=0 && r<n && str[l]==str[r])    // loop while l>=0, r<n and str[l]==str[r]

        {

            if(r-l+1 > end)    // check if r-l+1 > end

            {

                start=l;    // Assign start with value of l

                end=r-l+1;    // assign end with value of r-l+1

                

            }

            l--;    // decrement l

            r++;    // increment r

        }

 

        // odd substring    # odd substring

        l=i-1;    // Assign value of l as i-1

        r=i+1;    // Assign value of r as i+1

 

        while(l>=0 && r<n && str[l]==str[r])    // loop while l>=0, r<n and str[l]==str[r]

        {

            if(r-l+1 > end)    // check if r-l+1 > end

            {

                start=l;    // Assign start with value of l

                end=r-l+1;    // assign end with value of r-l+1

                

            }

            l--;    //decrement l

            r++;    // increment r

        }

    }

    return str.substr(start,end);    // return substring of str from start to end

}
